src/tests/system/tests/test_sssctl.py

```diff
@@ -242,6 +216,34 @@ def test_sssctl__check_missing_domain_name(client: Client):
     assert ex.match(r"Section \[domain\/\] is not allowed. Check for typos.*"), "Wrong error message was returned"
 
 
+@pytest.mark.tools
+@pytest.mark.topology(KnownTopology.Client)
+@pytest.mark.parametrize(
+    "contents,expected",
+    [
+        ("[domain/local]\ninvalid_option = True", "Attribute 'invalid_option' is not allowed"),
+        ("[invalid/local]\ninvalid_option = True", "Section [invalid/local] is not allowed"),
+    ],
+)
+def test_sssctl__check_invalid_option_name_in_snippet(client: Client, contents: str, expected: str):
+    """
+    :title: sssctl config-check validates configuration snippet
+    :setup:
+        1. Create a config snippet with an invalid option
+    :steps:
+        1. Check the configuration using sssctl
+    :expectedresults:
+        1. The config check fails with the appropriate output
+    :customerscenario: True
+    """
+    client.sssd.common.local()
+    client.fs.write("/etc/sssd/conf.d/01_snippet.conf", contents, mode="640")
```

Comment:
This migh become problematic with regards of ownership of the file and permissions being wildly different across different sssd versions.

Reply:
sssd-2.10+ currently chown/chmod-s config files in a service file. So mode doesn't matter much (as long as o+ isn't granted).
But this won't work for sssd-2.9, of course.

Reply:
What do you recommend in this case? Keep it as is, `mode="640"`?

Reply:
Just omit this arg.